plugin/src/test/kotlin/org/jlleitschuh/gradle/ktlint/KtlintPluginTest.kt

```diff
@@ -0,0 +1,77 @@
+package org.jlleitschuh.gradle.ktlint
+
+import org.gradle.testkit.runner.GradleRunner
+import org.gradle.testkit.runner.TaskOutcome
+
+import org.hamcrest.CoreMatchers.containsString
+import org.hamcrest.CoreMatchers.equalTo
+
+import org.junit.Assert.assertThat
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.rules.TemporaryFolder
+
+class KtlintPluginTest {
+
+    @get:Rule
+    val temporaryFolder = TemporaryFolder()
+
+    @Before
+    fun setupBuild() {
+        temporaryFolder.root.apply {
+            resolve("settings.gradle").writeText("")
+            resolve("build.gradle").writeText("""
+                plugins {
+                    id("org.jetbrains.kotlin.jvm") version "1.2.21"
+                    id("org.jlleitschuh.gradle.ktlint")
+                }
+
+                repositories {
+                    gradlePluginPortal()
+                }
+            """.trimIndent())
+            resolve("src/main/kotlin").mkdirs()
+        }
+    }
+
+    @Test
+    fun `check tasks`() {
```

Comment:
Maybe better to split it into two test cases:
- \`Should fail check on failing sources\`
- \`Should succeed check on clean sources\`

Reply:
Good point, done